index.html

```diff
@@ -2543,18 +2588,269 @@ <h1>
     </h1>
 
     <p>
-A <a>DID resolver</a> is a software or hardware component with an API for
-resolving <a>DIDs</a> of at least one <a>DID method</a>. It executes the read
-operation for the <a>DID method</a> corresponding to the <a>DID</a> being
-resolved to obtain the authoritative <a>DID document</a>. For more information,
-see Section <a href="#read-verify"></a>.
+This section defines the input and output contracts of the 
+<a>DID resolution</a> and <a>DID URL dereferencing</a> functions. The
+exact implementation of these functions is out of scope for this specification,
+but some considerations for implementors are available in [[?DID-RESOLUTION]].
     </p>
-
+    
     <p>
-The interfaces and algorithms for resolving <a>DIDs</a> and dereferencing
-<a>DID URLs</a> are specified in [[?DID-RESOLUTION]].
+All conformant <a>DID resolvers</a> MUST implement the <a>DID resolution</a> function
+for at least one <a>DID method</a> and be able to return a <a>DID document</a> in 
+at least one conformant representation. All conformant <a>DID URL dereferencers</a>
+MUST implement dereferencing for at least one conformant representation type.
     </p>
 
+    <section>
+        <h2>
+DID Resolution
+        </h2>
+        
+        <p>
+The <a>DID resolution</a> function resolves a <a>DID</a> into a <a>DID document</a>
+by using the "Read" operation of the applicable <a>DID method</a>. (See <a href="#read-verify"></a>.)
+The details of how this process is accomplished is outside the scope of this
+specification, but all implementations MUST implement a function in the form:
+        </p>
+        
+        <p>
+<code>resolve ( did, input-options ) -> ( did-document, did-document-metadata, did-resolution-metadata )</code>
+        </p>
+        
+        <p>
+The inputs of this function are a <a>DID</a> and a set of input options. The 
+<a>DID</a> is REQUIRED and MUST NOT include a fragment component. The input options
```

Comment:
> MUST NOT include a fragment component

Is there a need to specify this? For starters, `#` is not allowed as per the rules in the generic syntax.

Reply:
Yes, it's not technically needed here, but I think it at least warrants a note because of the lazy way people refer to a "DID URL" as a "DID". Being clear that it doesn't have a fragment is helpful.

Reply:
But then now I ask: why not also say that query parameters must also be excluded, or anything else that renders this input non-compliant with the generic syntax.

Anyway, I'm just pointing out words that, to me, distract from the meat of what this text is getting to. I don't really have a strong opinion on whether to include it or exclude it.

Reply:
I think it might make sense for it to be an inline note instead of a normative requirement for those reasons.